railties/lib/rails/generators/rails/db/system/change/change_generator.rb

```diff
@@ -113,19 +112,17 @@ def edit_compose_yaml
 
               compose_config = YAML.load_file(compose_yaml_path)
 
-              db_service_names.each do |db_service_name|
-                compose_config["services"].delete(db_service_name)
-                compose_config["volumes"]&.delete("#{db_service_name}-data")
-                compose_config["services"]["rails-app"]["depends_on"]&.delete(db_service_name)
+              Devcontainer::Database.all.each do |database|
+                compose_config["services"].delete(database.name)
+                compose_config["volumes"]&.delete(database.volume)
+                compose_config["services"]["rails-app"]["depends_on"]&.delete(database.name)
               end
 
-              db_service = db_service_for_devcontainer
-
-              if db_service
-                compose_config["services"].merge!(db_service)
-                compose_config["volumes"] = { db_volume_name_for_devcontainer => nil }.merge(compose_config["volumes"] || {})
+              if devcontainer_database.service
```

Comment:
Can we go a step further and also refactor all the other places that have different behavior depending on the databsae in the generator? There is no reason why this object should be devcontaienr specific. We need to find the gem for the database, the docker image for the database, etc.

Reply:
Yeah I can do that. The only slightly annoying thing is including all the other db that aren't available for devcontainer. I will work on it when I find some time this week.

Reply:
We can drop those databases. We already decided we would do, but never had a reason to. Now we have that reason.